### PR TITLE
feat: principal identity — system_role column, TaskOriginator, DB-driven OutboundGateway

### LINKED ISSUE #457: Refactor per-channel CEO identity fields into a CeoIdentity map
## Background

The current `OutboundGatewayConfig` carries CEO identity as flat fields:

```ts
ceoEmail?: string
ceoSignalNumber?: string
```

Each new outbound channel requires a new field on the config interface, a new branch in `isCeoRecipient()`, and a new branch in `sendEmailDraft()`. This is config proliferation — the gateway is accreting channel-specific knowledge that should live in a single, extensible structure.

## Proposed Refactor

Replace the per-channel fields with a `ceoIdentity: Map<string, string>` (channel → identifier):

```ts
// Before:
ceoEmail?: string
ceoSignalNumber?: string

// After:
ceoIdentity?: Map<string, string>
// e.g. { email: 'ceo@example.com', signal: '+14155551234' }
```

`isCeoRecipient(channel, address)` and the `sendEmailDraft()` CEO check both query `config.ceoIdentity.get(channel)` — no per-channel conditionals required.

`src/config.ts` loads this from a structured env var, e.g.:

```
CEO_IDENTITIES=email:ceo@example.com,signal:+14155551234
```

Adding a new channel (e.g. Slack, SMS) then requires only a new entry in `CEO_IDENTITIES`, not a new config field or gateway code branch.

## Acceptance Criteria

- [ ] `OutboundGatewayConfig` replaces `ceoEmail` and `ceoSignalNumber` with a single `ceoIdentity: Map<string, string>` (or similar)
- [ ] `isCeoRecipient()` and the `sendEmailDraft()` CEO check use the map instead of individual fields
- [ ] `src/config.ts` loads CEO identities from a structured config (e.g. `CEO_IDENTITIES=email:ceo@example.com,signal:+14155551234`)
- [ ] All existing tests continue to pass
- [ ] Adding a new channel no longer requires a new config field

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

- **`system_role` column on contacts** — adds `TEXT CHECK (system_role IN ('principal', 'agent'))` with partial unique indexes enforcing at most one principal and one agent. Separates system authorization designation from the free-text `role` field. Migration 035 backfills from existing `role = 'ceo'` and `role = 'agent'` data.
- **`TaskOriginator` replaces `ceoInitiated: boolean`** — the dispatcher stamps a richer originator on every task carrying `contactId`, `systemRole`, `channel`, and `initiatedAt`. `isPrincipalOriginated(metadata)` replaces all `ceoInitiated === true` checks. Originator survives task delegation, resolving the autonomy gate bug for scheduled CEO-authorized actions. Channel-supplied `originator` is stripped by `mergeTaskMetadata` (security).
- **OutboundGateway uses DB identity lookup** — removes flat `ceoEmail`/`ceoSignalNumber` config fields; loads principal contact's channel identities at startup. `isPrincipalRecipient()` replaces `isCeoRecipient()`.
- **Startup readiness check** — system refuses inbound messages if no `system_role='principal'` contact exists. Extensible `ReadinessCheck` interface for future setup gates.

## Files changed

- `src/db/migrations/035_add_system_role.sql` — migration
- `src/contacts/types.ts` — `SystemRole`, `TaskOriginator`, `systemRole` on `Contact`/`ResolvedSender`/`SenderContext`
- `src/contacts/contact-service.ts` — `system_role` in all queries, `findContactBySystemRole()`
- `src/contacts/principal.ts` — new `isPrincipalOriginated()` / `isAgentOriginated()` helpers
- `src/contacts/contact-resolver.ts` — CLI path uses `findContactBySystemRole('principal')`
- `src/contacts/ceo-bootstrap.ts` + `src/entity-context/bootstrap.ts` — set `system_role` on bootstrap
- `src/dispatch/dispatcher.ts` — `TaskOriginator` stamping, security stripping
- `src/skills/execution.ts` — elevated gate uses `isPrincipalOriginated`
- `src/skills/outbound-gateway.ts` — DB-driven principal identity check
- `src/agents/runtime.ts` — shows `systemRole` in sender context injection
- `src/contacts/confidence-pipeline.ts` — uses `systemRole` instead of `role`
- `src/startup/readiness.ts` — new readiness check runner
- `src/index.ts` — wires identity loading and readiness checks
- 5 skill handlers (send-draft, approve-action, deny-action, dismiss-action, list-pending-actions)

## Test plan

- [ ] 2024 unit tests pass (2 pre-existing infra failures requiring live DB are expected)
- [ ] `src/contacts/principal.test.ts` — `isPrincipalOriginated` / `isAgentOriginated` edge cases
- [ ] `tests/unit/startup/readiness.test.ts` — pass, fail, throw, empty cases
- [ ] `tests/unit/dispatch/dispatcher.test.ts` — originator stamped for principal/non-principal senders; hostile originator from channel metadata stripped
- [ ] `tests/unit/skills/execution.test.ts` — elevated gate passes for principal-originated, blocks for agent/null
- [ ] `tests/unit/skills/outbound-gateway.test.ts` — `isPrincipalRecipient` resolves from identity array
- [ ] Migration 035 verified unique — `ls src/db/migrations/ | sort` shows no prefix collision
- [ ] TypeScript: `tsc --noEmit` exits clean

## Related

Closes #457

## Summary by CodeRabbit

* **New Features**
  * Added startup readiness checks to ensure principal contact is configured before accepting inbound messages.
  * Implemented principal-originated task tracking to preserve authorization context across task delegation.

* **Bug Fixes**
  * Fixed scheduled authorized-action autonomy gate to properly recognize principal-originated tasks through delegation chains.

* **Improvements**
  * Consolidated principal identity into database-driven system role, separating authorization identity from user-defined roles.
  * Enhanced task metadata to include origin context (contact, system role, channel, timestamp) for better auditability.
  * Simplified outbound configuration by replacing email/signal number settings with unified principal identity management.

[Review Change Stack](https://app.coderabbit.ai/change-stack/josephfung/curia/pull/506)

### CODE REVIEW

**Comment by josephfung** [comment]
@coderabbitai review

File: skills/send-draft/handler.test.ts [hunk 1]
```diff
@@ -79,15 +88,15 @@ describe('SendDraftHandler', () => {
...
+    if (!result.success) expect(result.error).toMatch(/principal authorization/i);
   });
 
-  it('rejects when ceoInitiated is false', async () => {
-    const ctx = makeCtx({ taskMetadata: { ceoInitiated: false } });
+  it('rejects when originator.systemRole is not principal', async () => {
+    const ctx = makeCtx({ taskMetadata: { originator: { contactId: 'other-id', systemRole: 'agent', channel: 'cli', initiatedAt: new Date().toISOString() } } });
     const result = await handler.execute(ctx);
     expect(result.success).toBe(false);
   });
```
[the remark this answers is not in the document]
Comment (josephfung) [reply skills/send-draft/handler.test.ts:103]: Fixed in 21c8221. Added `expect(result.error).toMatch(/principal authorization/i)` to the 'rejects when originator.systemRole is not principal' test — it now pins the failure to the auth gate, not just any upstream guard.

File: src/contacts/contact-resolver.ts [hunk 2]
```diff
@@ -43,49 +43,50 @@ export class ContactResolver {
...
+        // Anything else (TypeError, programming bug in findContactBySystemRole, etc.) should propagate
         // so it is not silently misclassified as a transient DB blip.
         const isDbError = err !== null && typeof err === 'object' && 'code' in err;
         if (!isDbError) {
           throw err;
         }
         // DB error: log at error level so Sentry captures it, then fall through to synthetic ID.
         // The CLI continuing to work in degraded mode is acceptable; operators still need to know.
-        this.logger.error({ err }, 'contact-resolver: DB error looking up CEO contact — falling back to synthetic ID');
+        this.logger.error({ err }, 'contact-resolver: DB error looking up principal contact — falling back to synthetic ID');
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/contacts/contact-resolver.ts:82]: Fixed in 21c8221. Narrowed to a real SQLSTATE signature:
```ts
const sqlState = err !== null && typeof err === 'object' && 'code' in err
  ? (err as { code?: unknown }).code
  : undefined;
const isDbError = typeof sqlState === 'string' && /^[0-9A-Z]{5}$/.test(sqlState);
```
Five-character alphanumeric SQLSTATE codes are a pg-driver guarantee. AgentErrors and TypeErrors won't match.

File: src/index.ts [hunk 3]
```diff
@@ -703,6 +705,51 @@ async function main(): Promise<void> {
...
+  ];
+  const readinessReport = await runReadinessChecks(readinessChecks);
+
+  if (!readinessReport.ready) {
+    for (const failure of readinessReport.failures) {
+      logger.error({ check: failure.name, reason: failure.reason }, 'Startup readiness check failed');
+    }
+    throw new Error(`Startup readiness failed: ${readinessReport.failures.map((f) => f.name).join(', ')}`);
+  }
+  logger.info('All startup readiness checks passed');
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/index.ts:749]: Fixed in 21c8221 — removed the dead `if (contactService)` guard (ContactService.createWithPostgres() never returns undefined) and promoted `Contact` to the static import at line 45 alongside `ChannelIdentity`. The inline dynamic type import is gone.

File: src/skills/execution.ts [hunk 4]
```diff
@@ -298,22 +299,19 @@ export class ExecutionLayer {
...
+          { skillName, caller: caller ? { role: caller.role, channel: caller.channel } : null },
+          'Elevated skill blocked: task not originated by principal',
+        );
         return {
           success: false,
-          error: this.wrapSkillError(`Skill '${skillName}' requires elevated privileges — caller role '${caller.role ?? 'none'}' on channel '${caller.channel}' is not authorized`),
+          error: this.wrapSkillError(`Skill '${skillName}' requires elevated privileges — task was not originated by the principal`),
         };
       }
     }
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/skills/execution.ts:317]: Fixed in 21c8221. The approve-action re-invocation now passes `taskMetadata: ctx.taskMetadata` — the originator flows through, so the elevated gate passes when approve-action itself is principal-originated (which it always must be, per its own check at line 18). Without this, any elevated skill (scheduler-create, contact-grant-permission, etc.) would silently fail on re-execution through the approval flow.

File: tests/unit/skills/outbound-gateway.test.ts [hunk 5]
```diff
@@ -38,6 +38,27 @@ function createMocks() {
...
+    channel,
+    channelIdentifier,
+    label: null,
+    verified: true,
+    verifiedAt: null,
+    source: 'ceo_stated' as const,
+    createdAt: new Date(),
+    updatedAt: new Date(),
+  };
+}
```
[the remark this answers is not in the document]
Comment (josephfung) [reply tests/unit/skills/outbound-gateway.test.ts:60]: Fixed in 21c8221. Replaced all `ceoEmail: 'ceo@example.com'` and `ceoSignalNumber` constructor arguments with `principalIdentities: [makePrincipalIdentity('ceo@example.com')]` (35 occurrences). The two 'not configured' tests are updated to omit `principalIdentities` instead, with clarifying comments. The `expect(notificationEvent.payload.ceoEmail)` assertions at lines 272 and 1545 are untouched — `ceoEmail` is still a valid field on the bus event payload.

**Comment by josephfung** [comment]
Addressed all CodeRabbit findings in 21c8221. Summary by comment:

**Inline (5 actionable):**
- `send-draft/handler.test.ts:102` — added `toMatch(/principal authorization/i)` to the non-principal originator rejection test ✅
- `contact-resolver.ts:76` — narrowed `isDbError` to SQLSTATE pattern (`/^[0-9A-Z]{5}$/`) ✅
- `index.ts:751` — removed dead `if (contactService)` guard; `Contact` promoted to the static import ✅
- `execution.ts:317` — `approve-action` re-invocation now passes `taskMetadata: ctx.taskMetadata` so elevated skills pass the principal-origination gate when re-executed via approval ✅
- `outbound-gateway.test.ts:60` — all 35 `ceoEmail`/`ceoSignalNumber` constructor usages replaced with `principalIdentities: [makePrincipalIdentity('ceo@example.com')]`; 'not configured' test names/comments updated ✅

**Outside diff / nitpicks (all fixed):**
- `ceo-bootstrap.ts:164` — 23505 catch now inspects `err.constraint`; if it matches `idx_contacts_system_role_*` it logs a clear operator message and rethrows before the channel-identity re-query path ✅
- `entity-context/bootstrap.ts:75` — catch block now logs with `constraint` field when `idx_contacts_system_role_agent` fires ✅
- `send-draft/handler.ts:183` — human.decision publish moved inside an `else` branch; missing provenance logs an error and skips the publish entirely (no counterfeit audit rows) ✅
- `outbound-gateway.ts:1184` — `else` branch logs `error` with `blockId`/`draftId`/redacted recipient when no principal email is cached for the draft-block notification ✅
- `dispatcher.test.ts:1549,1607` — `if (originator)` guards replaced with unconditional `toBeDefined()` + `toBeNull()` assertions ✅
- `execution.test.ts:147` — added regression test: `role: 'ceo'` caller with no originator must fail the elevated gate ✅

**Skipped (genuinely low-value):**
- `contact-service.ts` warn on unexpected `system_role` — the DB `CHECK` constraint makes an out-of-range value impossible in normal operation. Not worth the IIFE noise per coercion site.
- `readiness.ts` per-check timeout — the single check is a single indexed DB query; startup latency budget covers it. Can slot in with the readiness framework expansion tracked under #392/#486.
